utils/cautils/acmeutils.go

```diff
@@ -494,7 +483,94 @@ func doDeviceAttestation(ctx *cli.Context, ac *ca.ACMEClient, ch *acme.Challenge
 	if err != nil {
 		return fmt.Errorf("error marshaling payload: %w", err)
 	}
-	return ac.ValidateWithPayload(ch.URL, payload)
+
+	ui.Printf("Using Device Attestation challenge to validate %q", identifier)
+	ui.Printf(" .") // Indicates passage of time.
+
+	if err := ac.ValidateWithPayload(ch.URL, payload); err != nil {
+		ui.Printf(" Error!\n\n")
+		return errors.Wrapf(err, "error validating ACME Challenge at %s", ch.URL)
+	}
+
+	durationBetweenAttempts := 2 * time.Second
+	if err := getChallengeStatus(ac, ch, durationBetweenAttempts); err != nil {
+		ui.Printf(" Error!\n\n")
+		return err
+	}
+
+	ui.Printf(" done!\n")
+	return nil
+}
+
+// getChallengeStatus retrieves the ACME Challenge status from the CA. It
+// will retry this 10 times if the status is not `invalid` or `valid`. This
+// is the case, for example, when the CA hasn't been able to connect to the
+// HTTP endpoint for some reason.
+func getChallengeStatus(ac *ca.ACMEClient, ch *acme.Challenge, durationBetweenAttempts time.Duration) error {
+	var (
+		isValid = false
+		vch     *acme.Challenge
+		err     error
+	)
+	for attempts := 0; attempts < 10; attempts++ {
+		vch, err = ac.GetChallenge(ch.URL) // TODO(hs): GetChallenge should return an ACME GetChallenge client response type; not core acme.Challenge type (for safety)
+		if err != nil {
+			return errors.Wrapf(err, "error retrieving ACME Challenge at %s", ch.URL)
+		}
+		// break early if challenge validation failed. Due to nature of the `device-attest-01`
+		// challenge validation, this will (currently) happen immediately, as the validation is
+		// performed synchronously by the CA when the request for `ac.ValidateWithPayload` is
+		// sent. For `http-01`, this is the case when the key authorization doesn't match.
+		// In the (near) future we might change the way the validation is performed, so keeping
+		// the retry logic intact for now.
+		if vch.Status == "invalid" {
+			break
+		}
+		if vch.Status == "valid" {
+			isValid = true
+			break
+		}
+		ui.Printf(".")
+		time.Sleep(durationBetweenAttempts) // NOTE: a bit shorter than the HTTP challenge type
```

Comment:
The comment does not make sense now.

```suggestion
		time.Sleep(durationBetweenAttempts)
```